agent/tags.go

```diff
@@ -157,14 +157,24 @@ func (t *tagFetcher) Fetch(l logger.Logger, conf FetchTagsConfig) []string {
 
 	// Attempt to add the default GCP meta-data tags
 	if conf.TagsFromGCPMetaData {
-		gcpTags, err := t.gcpMetaDataDefault()
-		if err != nil {
-			// Don't blow up if we can't find them, just show a nasty error.
-			l.Error(fmt.Sprintf("Failed to fetch Google Cloud meta-data: %s", err.Error()))
-		} else {
-			for tag, value := range gcpTags {
-				tags = append(tags, fmt.Sprintf("%s=%s", tag, value))
+		l.Info("Fetching GCP meta-data...")
+
+		err := retry.Do(func(s *retry.Stats) error {
+			gcpTags, err := t.gcpMetaDataDefault()
+			if err != nil {
+				// Don't blow up if we can't find them, just show a nasty error.
+				l.Error(fmt.Sprintf("Failed to fetch Google Cloud meta-data: %s", err.Error()))
```

Comment:
Nitpick, but if you used GCP above, use it here too.
```suggestion
				l.Error(fmt.Sprintf("Failed to fetch GCP meta-data: %s", err.Error()))
```